tests/test_scheduler.py

```diff
@@ -684,16 +734,16 @@ def test_scheduler_e2e_matmul(a, b):
     i, j, k = Field("i"), Field("j"), Field("k")
 
     plan = Plan(
-        [
+        (
             Query(Alias("A"), Table(Literal(a), (i, k))),
             Query(Alias("B"), Table(Literal(b), (k, j))),
             Query(Alias("AB"), MapJoin(Literal(mul), (Alias("A"), Alias("B")))),
             Query(
                 Alias("C"),
-                Reorder(Aggregate(Literal(add), Literal(0), Alias("AB"), (k,)), (i, j)),
+                Aggregate(Literal(add), Literal(0), Alias("AB"), (k,)),
```

Comment:
Reorder prevents "propagate queries backwards" optimization.

Reply:
Solved in https://github.com/finch-tensor/finch-tensor-lite/pull/124 - I can revert that change.